src/core/midi/MidiApple.cpp

```diff
@@ -0,0 +1,494 @@
+/*
+ * MidiApple.cpp - Apple raw MIDI client
+ *
+ * Copyright (c) 2005-2015
+ *
+ * This file is part of LMMS - http://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+
+#include "MidiApple.h"
+
+#include <QLabel>
+#include <QLineEdit>
+#include <algorithm>
+
+#include "ConfigManager.h"
+#include "gui_templates.h"
+#include "MidiPort.h"
+#include "Note.h"
+
+
+
+#ifdef LMMS_BUILD_APPLE
+
+#include <CoreMIDI/CoreMIDI.h>
+
+#define SYSEX_LENGTH 1024
+
+#define MIN(a,b) \
```

Comment:
This may be best suited for [`lmms_math.h`](https://github.com/LMMS/lmms/blob/master/include/lmms_math.h)